Cattac/Cattac/GameEngine.swift

```diff
@@ -26,14 +26,17 @@ class GameEngine {
 
     // The AI engine that is used when multiplayer mode is not active
     private var gameAI: GameAI!
+    
+    /// States to advance, initialized at 1 to rollover PreStart state.
+    private var statesToAdvance: Int = 1
 
     var gameManager: GameManager = GameManager()
     
     /// Player index (we should change to player-id instead).
     var playerNumber = 1
     
-    // The initial game state is to be set at precalculation
-    var state: GameState = .Precalculation
+    // The initial game state is to be set at Initialization
+    var state: GameState = .PreStart
```

Comment:
There might be a better way to implement, rationale is for the following code to run on first go.
```
 func gameLoop() {
        if statesToAdvance > 0 {
            advanceState()
        } else {
            // No need to execute state methods if state unchanged
            return
        }
        ...
```

While we probably don't want to move this if-else to the end of `gameLoop()` since that will still run the state methods 60 times per second, and we must run `precalculate()` for `reachableNodes`